src/main/java/frc/robot/commands/DefaultTeleop.java

```diff
@@ -88,19 +85,21 @@ public void execute() {
         double translationVal = MathUtil.applyDeadband(yAxis, Constants.stickDeadband);
         double strafeVal = MathUtil.applyDeadband(xAxis, Constants.stickDeadband);
         double rotationVal;
-        
-        if (driver.getRawButtonReleased(XboxController.Button.kB.value)) {
-            isAligning = true;
-        } else if (MathUtil.applyDeadband(rotationAxis, Constants.stickDeadband) > 0) {
-            isAligning = false;
-        }
 
-        if ((MathUtil.applyDeadband(rotationAxis, Constants.stickDeadband) > 0 && isAligning) || (bButtonPressed && isAligning)) {
+        if ((isAligning && MathUtil.applyDeadband(rotationAxis, Constants.stickDeadband) != 0) || isAligning && ((bButtonPressed || xButtonPressed || yButtonPressed || aButtonPressed))) {
             isAligning = false;
-        } else if (bButtonPressed && isAligning == false) {
+        } else if (!isAligning && (bButtonPressed || xButtonPressed || yButtonPressed || aButtonPressed)) {
             isAligning = true;
         }
 
+        if (bButtonPressed) {
+            AlignPosition.setPosition(AlignPosition.AmpPos);
+        } else if (aButtonPressed) {
+            AlignPosition.setPosition(AlignPosition.ClimbPos);
+        } else if (xButtonPressed) {
```

Comment:
We really only need to call setPosition once per button press. After that, the button variable can be set back to false. I recommend moving this logic into a separate method (along with the button updating logic) to make this cleaner.

Reply:
removed already in previous comment... hehe